services/species.service.js

```diff
@@ -0,0 +1,36 @@
+const { v4: uuid } = require('uuid');
+const SpeciesModel = require('../models/species.schema');
+
+const getAllSpecies = async () => {
+  return SpeciesModel.find({});
+};
+
+const changeListing = async (speciesId) => {
+  const changedListing = !SpeciesModel.findById(speciesId).isAssigned;
+  return SpeciesModel.findByIdAndUpdate(speciesId, { isAssigned: changedListing });
+};
+
+const addNewSpecies = async (species) => {
+  if (!species.name || !species.code) {
+    throw new Error('Arguments missing in species');
+  }
+  const newSpecies = new SpeciesModel({
+    _id: uuid(),
```

Comment:
You actually don't need to do this manually, [mongo will generate it for you](https://docs.mongodb.com/manual/reference/glossary) if you just remove it here and remove it from the schema.